src/charm.py

```diff
@@ -230,32 +246,51 @@ def self_scraping_job(self):
 
         return [job]
 
+    def _update_blackbox_config_yaml_given_dict_from_relation(self, modules: dict) -> None:
+        """Update the blackbox config yaml given a dict of modules defined in relation.
+
+        This function takes the dict of modules from the BlackboxExporterRequirer and
+        updates the config file with the required modules, only if they do not exist already
+        in the config file.
+
+        Raises:
+            yaml.YAMLError: If there is an error in the YAML formatting or parsing.
+            TypeError: If type is not a string, as `yaml.safe_load` requires a string input.
+        """
+        if not modules:
+            return
+
+        config_file_data = self.container.pull(self._config_path).read()
+
+        if not config_file_data:
+            return
```

Comment:
- Are we guaranteed to always have a file in `_config_path`? And if it doesn't exists we'd get an (unhandled) exception from ops?
- Seems like the config file is written only later, and the probes won't be rendered there until the next hook. I.e. seems like an implicit dependency on `update_status`?

Reply:
improved in 8eaaac8170f5fdc10c1d2483b154b262fa145ead

Reply:
added comment in 734f467b0b1333a705578a954fc0eb14f3b4e04b

Reply:
See [comment](https://github.com/canonical/blackbox-exporter-k8s-operator/pull/43#issuecomment-2618371884)